packages/flutter/lib/src/widgets/app.dart

```diff
@@ -555,11 +555,15 @@ class WidgetsApp extends StatefulWidget {
   /// This list must not null. Its default value is just
   /// `[const Locale('en', 'US')]`.
   ///
-  /// The order of the list matters. By default, if the device's locale doesn't
-  /// exactly match a locale in [supportedLocales] then the first locale in
-  /// [supportedLocales] with a matching [Locale.languageCode] is used. If that
-  /// fails then the first locale in [supportedLocales] is used. The default
-  /// locale resolution algorithm can be overridden with [localeResolutionCallback].
+  /// The order of the list matters. The default locale resolution algorithm, 
+  /// [basicLocaleListResolution], uses a variation of the truncation algorithm where
+  /// it attempts to match by all subtags, language+script, language+country,
+  /// language-only, and the [supportedLocales.first] as a fallback. When more
+  /// than one supported locale matches one of these criteria, only the first one
+  /// is used.
+  ///
+  /// The default locale resolution algorithm can be overridden with [localeListResolutionCallback]
+  /// and/or [localeResolutionCallback].
```

Comment:
Should we really mention localeResolutionCallback here? My understanding was that localeListResolutionCallback is the preferred one. If we need to mention both, we should maybe make it clear which one or how to chose.

Also, an chance you can add a sentence or two describing why one would override the default one?

Reply:
Yeah, makes sense, since we prefer the list version and the old one is just there for legacy reasons, I should probably just remove it.